### ISSUE TITLE
How do we deal with translations?

### ISSUE THREAD

**Issue by ladyleet** [ReactiveX/rxjs-docs#29]
I've been thinking about this and I actually like the way Vue does this best.

Check out the screenshot. 

<img width="1185" alt="screenshot 2017-10-07 04 01 26" src="https://user-images.githubusercontent.com/8270563/31307221-46858b8c-ab14-11e7-9d3c-b775427bede4.png">

**Comment by sculove** [comment]
We need an i18n development environment.
Angular provides i18n. https://angular.io/guide/i18n
There is also a library like ngx-translate. http://www.ngx-translate.com/

The way angular guides is actually cumbersome.
I don't know what actually would be good.
Do you have any idea?

**Comment by SangKa** [comment]
Vue use different sites for each language.
Maybe we can use the experience of what facebook does, just like jest or react.
Can check [this page](http://facebook.github.io/jest/), they use the [crowdin](https://crowdin.com/project/jest) for i8in.

**Comment by sculove** [comment]
@SangKa the crowdin is good.
I tried a different idea.
a crowdin and google translate provide `xliff` format.
I think If we use Google Translate API, we can easily create a document draft for each language.

I found this issue.
https://github.com/angular/angular/issues/14145

**Comment by jonguk0114** [comment]
I have not used translation tools or platforms but  just took a look at [crowdin](https://crowdin.com).

Crowdin provides multiple translators with features for collaboration such as discussion, voting, adding terms for translation from original language to translated language, suggestion, and etc. 

I believe that crowdin is helpful to complete consistent translation by discussing or voting about different opinions. 

According to [Crowdin Open Source Request From](https://crowdin.com/page/open-source-project-setup-request), it is free of charge after open source project request is approved.
 
I think crowdin is good too.

**Comment by Lakston** [comment]
I use ngx-translate at work and it's working really well for us, we have one json file per language and use for example `{{ 'OPERATORS.GROUPED_OBSERVABLE.FORKJOIN.whatever_par_of_the_page | translate }}`

We would have to structure it really well so it doesn't become a mess after a while, separating each page in manageable chunks that are properly named could be quite long to do.

**Comment by ladyleet** [comment]
Hey all! :) Thanks for all the conversation around this! 

I was talking to Rahul from the Vue core team about their docs since I like those best from an accessibility standpoint. 

@chrisvfritz it would actually be super helpful to get your opinion on the docs and how we should best approach this.

@SangKa great to hear you have translated @btroncone's already! That will give us a very good head start in the Korean translation docs!

If you all have not yet done so, please send me an email to tracy@thisdot.co to be included in the learning team meeting on Monday 10/16.

**Comment by ladyleet** [comment]
so, i literally have *no* idea how to start with translations except for the idea that it *seems* like we need both NGX Translate AND something like Git Localize. Is this the case? :D

**Comment by znck** [comment]
Adding @kazupon & @phanan to the thread.

**Comment by phanan** [comment]
Below is our current flow for translating vuejs.org to [the Vietnamese version](https://vi.vuejs.org). We've just started, so the flow is very open to changes/suggestions and in no way the most optimized. Also, again since we're at the beginning of the translation process, our current focus is for the _untranslated documents_, which is a decisive factor for the flow. Anyway:

*

**Comment by phanan** [comment]
Thanks for the mention @znck.

Below is our current flow for translating vuejs.org to the Vietnamese version. We've just started, so the flow is very open to changes/suggestions and in no way the most optimized. Also, again since we're at the beginning of the translation process, our current focus is for the untranslated documents, which is a decisive factor for the flow. Anyway:

* The [i18n repo](https://github.com/vuejs-vn/vuejs.org/) is a direct folk from the English repo
* Each PR involves only ONE file, however small it is
* Before translating, a member would check the [PR](https://github.com/vuejs-vn/vuejs.org/pulls?utf8=%E2%9C%93&q=is%3Apr) page to make sure if it's not already been/being translated
* Create a new branch from `master` with the branch name being the _path_ to the file relatively to the root, e.g. `src/v2/guide/instance.md`
* Submit this branch as a PR early in the process, with the PR title being `[WIP] src/v2/guide/instance.md` following the previous example. This helps maintainers from others to decide what to do (or not) with the PR and the involved file.
* Continue translating and pushing commits to the branch
* Once translation is done, remove `[WIP]` from the title and tag the reviewers

We're not using any external tool atm (maybe we should?) Again, this flow is no way standard, and actually we're looking out for suggestions.

**Comment by ladyleet** [comment]
Does someone want to champion our process for the translations in Rx? We def need a champion who will orchestrate the process but obviously requires some man hours. :)

**Comment by kazupon** [comment]
@znck Thank you for your mention. 😺 

I'm translating Vue.js documents together with Vue.js Japan :jp: users  daily for Japanese to maintain the quality of translation.

From the experience of the translation work, I recommend using translation services like crowdin and gitlocalize. It's very difficult to track the diff of original document with human eyes 👀 .

I don't know crowdin due to not used crowdin, [gitlocalize](https://gitlocalize.com) provides the tracking feature.

If you don't use these services, we recommend that you introduce the translation flow with mentioned @phanan , or automatically pick up diff from original document with bot 🤖 .

for japanese translation, We use [this bot](https://github.com/re-fort/che-tsumi).

**Comment by ladyleet** [comment]
thank you @kazupon! again, anyone who is interested in helping drive the complete initiative let me know.

**Comment by chrisvfritz** [comment]
One other note: for documentation that's frequently updating and translated to many languages like the Vue core docs, having separate repos for each translation is very useful. If we had everything under a single repo, maintainers would have to ignore most PRs because they'd be unrelated to the source material.

**Comment by ladyleet** [comment]
@chrisvfritz that is super useful feedback chris. thanks!

**Comment by kwonoj** [comment]
So there seems few things we need to figure out (from contents logistics to actual impl) - let's try to list them and separate as each issue to track. I'll try to update them shortly.

**Comment by sotayamashita** [comment]
@ladyleet 

> I don't know crowdin due to not used crowdin, gitlocalize provides the tracking feature.

Hi, I am an engineer for [GitLocalize](https://gitlocalize.com/). Please feel free to ask me if you have any questions. I also have translated docs ever. I can help you even though you would not use GitLocalize. Thanks :bow:

**Comment by kwonoj** [comment]
@sotayamashita cool, let me peek those into too!

**Comment by Lakston** [comment]
Gitlocalize looks pretty awesome but reading [the docs](https://docs.gitlocalize.com/file_formats.html) it only deals with `.md` and static `.html` files. 

Digging into [ssr.vuejs.org](https://ssr.vuejs.org/) they seems to be using [GitBook](https://www.gitbook.com/) + [GitLocalize](https://gitlocalize.com/) and it looks easy to maintain and manage.

I use ngx-translate every day but for this project it looks like going the [ssr.vuejs.org](https://ssr.vuejs.org/) would be a better fit.

**Comment by sotayamashita** [comment]
@Lakston 

> Gitlocalize looks pretty awesome but reading the docs it only deals with .md and static .html files.

I appreciate you check our product out. What kind of would like to use?

> edit : If I have some time tonight (GMT) I'll start a personnal repo with GitBook and GitLocalize to see if I can understand how to set it up.

If you have question, please feel free to ask me via here or [@sota0805](https://twitter.com/sota0805).

**Comment by kwonoj** [comment]
note purpose: https://pontoon.mozilla.org/

**Comment by sotayamashita** [comment]
@kwonoj

 I've never heard of it and it looks cool. I can improve GitLocalize based on pontoon. I am wondering how to store translation data.

**Comment by kwonoj** [comment]
I'm not debating any particular at this moment, just left as memo that I don't forget. I think probably need to run couple of evalution to compare how those experiences are between way we listed here.

**Comment by kwonoj** [comment]
I'm making this is meta-issue to track general discussions, and created separate issue for discussing tool/pipeline at https://github.com/ReactiveX/rxjs-docs/issues/36 .

**Comment by getDanArias** [comment]
I would like to put on the table [**transifex**](https://www.transifex.com) as an option for managing the translation process. 

From their [site](https://docs.transifex.com/faq/all#what-is-considered-an-open-source-project): 

> Transifex offers an Open Source Program to Open Source projects that have no funding, revenue, and/or commercialization model. The intent is to support small and independent Open Source projects. 

Habitica, the game app, has been using it to great success for the world wide translation of their platform. The tracking system is outstanding! 

Here's some information on their integration with Github: [Using Transifex with GitHub in Your Development Workflow](https://docs.transifex.com/integrations/github/)

**Comment by kwonoj** [comment]
@getDanArias added in #36.

**Comment by feloy** [comment]
What about translating static texts, like site title, menu labels, etc. Do you want to use the internal Angular i18n?

I can work on the toolchain for this part.

**Comment by ladyleet** [comment]
@feloy we are chatting about this in #36

**Comment by xlozinguez** [comment]
I obviously will take care of the French whenever we have the tooling ready. Happy to partake in that effort also (cc @kwonoj & @ladyleet)
